app/models/document_storage.rb

```diff
@@ -0,0 +1,17 @@
+class DocumentStorage < MongoidBase
+  belongs_to :document
+
+  field :type, type: String
+
+  def read
```

Comment:
These three methods (`read`, `write`, and `search_content`) don't need to be here because we'll be overloading them in the specific type (EG html) anyways.  Not like a java interface.